dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/AccessLogFilter.java

```diff
@@ -68,141 +67,134 @@
-                                if (!now.equals(last)) {
-                                    File archive = new File(file.getAbsolutePath() + "." + last);
-                                    file.renameTo(archive);
-                                }
-                            }
-                            FileWriter writer = new FileWriter(file, true);
-                            try {
-                                for (Iterator<String> iterator = logSet.iterator();
-                                     iterator.hasNext();
-                                     iterator.remove()) {
-                                    writer.write(iterator.next());
-                                    writer.write("\r\n");
-                                }
-                                writer.flush();
-                            } finally {
-                                writer.close();
-                            }
-                        } catch (Exception e) {
-                            logger.error(e.getMessage(), e);
+    private void log(String accessLog, AccessLogData accessLogData) {
+        Set<AccessLogData> logSet = logQueue.computeIfAbsent(accessLog, k -> new ConcurrentHashSet<>());
+
+        if (logSet.size() < LOG_MAX_BUFFER) {
+            logSet.add(accessLogData);
+        } else {
+            //TODO we needs use force writing to file so that buffer gets clear and new log can be written.
+            logger.warn("AccessLog buffer is full skipping buffer ");
+        }
+    }
+
+    private void writeLogToFile() {
+        if (!logQueue.isEmpty()) {
+            for (Map.Entry<String, Set<AccessLogData>> entry : logQueue.entrySet()) {
+                try {
+                    String accessLog = entry.getKey();
+                    Set<AccessLogData> logSet = entry.getValue();
+                    if (ConfigUtils.isDefault(accessLog)) {
+                        processWithServiceLogger(logSet);
+                    } else {
+                        File file = new File(accessLog);
+                        createIfLogDirAbsent(file);
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("Append log to " + accessLog);
                         }
+                        renameFile(file);
+                        processWithAccessKeyLogger(logSet, file);
                     }
+
+                } catch (Exception e) {
+                    logger.error(e.getMessage(), e);
                 }
-            } catch (Exception e) {
-                logger.error(e.getMessage(), e);
             }
         }
     }
 
-}
+    private void processWithAccessKeyLogger(Set<AccessLogData> logSet, File file) throws IOException {
+        FileWriter writer = new FileWriter(file, true);
+        try {
+            for (Iterator<AccessLogData> iterator = logSet.iterator();
+                 iterator.hasNext();
+                 iterator.remove()) {
```

Comment:
need formation here.

Reply:
Here we have list of access log, which we are writting and then removing entry.

If you would provide more specific question then may be I can answer those.